src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+   *
+   * <ul>
+   *   <li>Remove a statement, represented by null
+   *   <li>Replace the right hand side expression with {@code 0}, {@code false}, or {@code null}
+   *   <li>Replace right hand side by a calculated value obtained from a passing assertion
+   *   <li>Remove the left hand side of a statement, retaining only the expression on the right
+   * </ul>
+   *
+   * Assertions are never simplified.
+   *
+   * @param currStmt statement to simplify
+   * @param primitiveValues map of primitive variable names to expressions representing their values
+   * @return list of statements, where each is a possible simplification of {@code currStmt}
+   */
+  private static List<Statement> getStatementReplacements(
+      Statement currStmt, Map<String, String> primitiveValues) {
+    List<Statement> replacements = new ArrayList<Statement>();
+
+    // Null represents removal of the statement.
+    replacements.add(null);
+
+    if (currStmt instanceof ExpressionStmt) {
+      Expression exp = ((ExpressionStmt) currStmt).getExpression();
+      if (exp instanceof VariableDeclarationExpr) {
+        // Create and return a list of possible replacement statements.
+        VariableDeclarationExpr vdExpr = (VariableDeclarationExpr) exp;
+
+        // Simplify right hand side to zero equivalent value, 0, false, or null.
+        replacements.add(rhsAssignZeroValue(vdExpr));
+
+        // Simplify right hand side to a value that was previously found in a passing assertion.
+        Statement rs = rhsAssignValueFromPassingAssertion(vdExpr, primitiveValues);
+        if (rs != null) {
+          replacements.add(rs);
+        }
+
+        // Simplify statement by removing the left hand side.
+        replacements.add(removeLeftHandSideSimplifiation(vdExpr));
+      }
+    }
+    return replacements;
+  }
+
+  /**
+   * If {@code currStmt} is a statement that is a passing assertion, store the value associated with
+   * the variable in the {@code primitiveValues} map.
+   *
+   * @param currStmt a statement
+   * @param primitiveValues a map of variable names to variable values, modified if {@code currStmt}
+   *     is a passing assertion, asserting a variable's value.
+   */
+  private static void storeValueFromAssertion(
+      Statement currStmt, Map<String, String> primitiveValues) {
+    // Check if the statement is an assertion regarding a value that can be used
+    // in a simplification later on.
+    if (currStmt instanceof ExpressionStmt) {
+      Expression exp = ((ExpressionStmt) currStmt).getExpression();
+      if (exp instanceof MethodCallExpr) {
+        MethodCallExpr mCall = (MethodCallExpr) exp;
+        // Check if the method call is an assertTrue statement
+        if (mCall.getName().equals("assertTrue")) {
+          // Is an assertTrue statement.
+          List<Expression> mArgs = mCall.getArgs();
+          if (mArgs.size() == 1) {
+            // Retrieve and store the value associated with the variable in the assertion.
```

Comment:
The minimizer now "checks" syntax by checking if the expression contains the string "==", but this doesn't seem like the ideal way since it seems like an expression with (a==b) != (c==d) could appear.